### PR TITLE
solfege: fix build

### PR DESCRIPTION
**Pull request by anthonyroussel**
###### Description of changes

* Fix failing build
* Replace deprecated `gpl3` license by `gpl3Only`
* Fix bad `meta.homepage` URL

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/misc/solfege/default.nix [hunk 1]
```diff
@@ -2,15 +2,16 @@
...
 
 buildPythonApplication rec {
   pname = "solfege";
   version = "3.23.4";
 
   src = fetchurl {
-    url = "mirror://sourceforge/solfege/solfege-${version}.tar.gz";
-    sha256 = "0sc17vf4xz6gy0s0z9ghi68yskikdmyb4gdaxx6imrm40734k8mp";
+    url = "http://alpha.gnu.org/gnu/solfege/solfege-${version}.tar.gz";
+    sha256 = "sha256-t6JJxgGk5hpN76o9snxtM07tkYnwpQ808M/8Ttw+gWk=";
```
Comment (bjornfor) [inline pkgs/misc/solfege/default.nix:14]: Why is there a hash change if the version is unchanged?

**Comment by bjornfor** [comment]
IMHO, the description of change in the PR message is also good to have in the commit.

**Comment by anthonyroussel** [comment]
> IMHO, the description of change in the PR message is also good to have in the commit.

You're right :)
I have splitted these different changes in multiple commits to make it more understandable.

File: pkgs/misc/solfege/default.nix (hunk 1 above)
Comment (anthonyroussel) [reply pkgs/misc/solfege/default.nix]: Yes indeed, the hash should not change.
Fixed

**Review by bjornfor — APPROVED** [review]

**Pull request merged**